generator/actions_core.ml

```diff
@@ -264,7 +264,7 @@ this function fails and the C<errno> is set to C<EINVAL>.|} };
 
   { defaults with
     name = "add_drive"; added = (0, 0, 3);
-    style = RErr, [String (PlainString, "filename")], [OBool "readonly"; OString "format"; OString "iface"; OString "name"; OString "label"; OString "protocol"; OStringList "server"; OString "username"; OString "secret"; OString "cachemode"; OString "discard"; OBool "copyonread"; OInt "blocksize"];
+    style = RErr, [String (PlainString, "filename")], [OBool "readonly"; OString "format"; OString "iface"; OString "name"; OString "label"; OString "protocol"; OStringList "server"; OString "username"; OString "secret"; OString "cachemode"; OString "discard"; OBool "copyonread"; OInt "blocksize"; OString "query"];
```

Comment:
This can't work as an `OString`, since a query is made up of a list of (key, value) elements.  This also leads to the confusion in this pull request around URI escaping.

The closest we have is `OStringList` (list of key=value fields) which might work.  Ideally we'd have `OHashtable` (analogous to `RHashtable`) but that doesn't exist, and adding it would be a very significant amount of work.

Quoting URLs in general is quite problematic. @berrange any thoughts on whether `OStringList` can be used safely here?